x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go

```diff
@@ -53,9 +54,10 @@ type Dimension struct {
 
 // Config holds a configuration specific for cloudwatch metricset.
 type Config struct {
-	Namespace  string      `config:"namespace" validate:"nonzero,required"`
-	MetricName string      `config:"metricname"`
-	Dimensions []Dimension `config:"dimensions"`
+	Namespace           string      `config:"namespace" validate:"nonzero,required"`
+	MetricName          string      `config:"metricname"`
+	Dimensions          []Dimension `config:"dimensions"`
+	ResourceTypeFilters []string    `config:"resource_type_filters"`
```

Comment:
do you expect this to hold more than one type? I cannot think of any example, but may be missing something

Reply:
Yeah, in theory it can hold more than one type https://docs.aws.amazon.com/cli/latest/reference/resourcegroupstaggingapi/get-resources.html. For example under `ec2`, there can be `ec2:instance` or `ec2:vpc` or `ec2:security-group`.  But in our case, when querying cloudwatch metrics, we are only getting metrics for ec2 instances. So here we only need `string` instead of `[]string`.